webapp/packages/core-connections/src/ConnectionNavigatorViewSettings.ts

```diff
@@ -31,8 +34,34 @@ export const CONNECTION_NAVIGATOR_VIEW_SETTINGS: Record<NavigatorView, Navigator
   },
 };
 
-export function isSimpleNavigatorView(settings: NavigatorSettingsInput): boolean {
-  const simple = CONNECTION_NAVIGATOR_VIEW_SETTINGS.simple;
-  return !Object.keys(simple)
-    .some(key => settings[key as keyof NavigatorSettingsInput] !== simple[key as keyof NavigatorSettingsInput]);
+export function getNavigatorView(settings: NavigatorSettingsInput): NavigatorView {
+  const isSimple = isNavigatorViewSettingsEqual(settings, CONNECTION_NAVIGATOR_VIEW_SETTINGS.simple);
+  const isAdvanced = isNavigatorViewSettingsEqual(settings, CONNECTION_NAVIGATOR_VIEW_SETTINGS.advanced);
+
+  if (isSimple) {
+    return 'simple';
+  }
+
+  if (isAdvanced) {
+    return 'advanced';
+  }
+
+  return 'custom';
+}
+
+export function isNavigatorSettingEnabled(
+  setting: NavigatorViewSettingsKeys, settings: NavigatorSettingsInput
+): boolean {
+  return settings[setting];
+}
+
+function isNavigatorViewSettingsEqual(settings: NavigatorSettingsInput, settingsToCompare: NavigatorSettingsInput) {
+  return !(Object.keys(settingsToCompare) as NavigatorViewSettingsKeys[])
+    .some(key => {
+      // we need to exclude it for now, cause we haven't implement custom navigator view logic yet
+      if (key === 'showSystemObjects') {
+        return false;
+      }
```

Comment:
just exclude this option from `CONNECTION_NAVIGATOR_VIEW_SETTINGS` object (simple or advanced is object with optional NavigatorSettingsInput keys)